### PR TITLE
Script for Mashing Up Doc Index Page

### PR DESCRIPTION
**Pull request by chidozieononiwu**

### CODE REVIEW

File: eng/pipelines/templates/scripts/mashup-doc-index.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        import os
+        import logging
+        import re
+        import shutil
+        from io import open
+
+        SITE_INDEX = r'${{ parameters.SourceDirectory }}\docfx_project\_site'
+        TOC_HTML_REGEX = r"\.\./toc.html"
+        NAV_TOC_HTML_REGEX = r"api/"
+        PREV_DIR_REGEX = r"\.\./"
```
Comment (scbedd) [inline eng/pipelines/templates/scripts/mashup-doc-index.yml]: This assumes that any and all `one directory up` indicators must be replaced. Given that this is generating an index, and no contents of packages (like readmes, other than the the main readme), is this a safe assumption?

Aka, This won't replace any occurrences of `..` in a code snippet?

File: eng/pipelines/templates/scripts/mashup-doc-index.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            content = re.sub(NAV_TOC_HTML_REGEX, '', content)
+            return content
+
+        if __name__ == "__main__":
+            html_files = locate_htmlfiles(os.path.join(SITE_INDEX, 'api'))
+
+            print(SITE_INDEX)
+            navtoc_location = os.path.join(SITE_INDEX, 'toc.html')
+
+            print(navtoc_location)
```
Comment (scbedd) [inline eng/pipelines/templates/scripts/mashup-doc-index.yml]: Mix between `print` and `logging`. It's not a big deal, but try to choose one and keep it consistent. (logging is definitely preferred)

File: eng/pipelines/templates/scripts/mashup-doc-index.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+            # Process the main toc.html and rename it to navtoc.html
+            try:
+                logging.info(
+                    "Process {}.".format(navtoc_location)
+                )
+                with open(navtoc_location, "r") as navtoc_stream:
+                    navtoc_content = navtoc_stream.read()
+                new_navtoc_content = process_navtoc(navtoc_content)
+                print("Process {}.".format(navtoc_content))
```
Comment (scbedd) [inline eng/pipelines/templates/scripts/mashup-doc-index.yml]: print vs log again. Is the content dumping for debug?

File: eng/pipelines/templates/scripts/mashup-doc-index.yml (hunk 1 above)
Comment (chidozieononiwu) [reply eng/pipelines/templates/scripts/mashup-doc-index.yml]: Yup I believe it is. The only other things in the code are directory names which don't contain slashes. There should be no case where there is an exception as long as this is used for just the index.

File: eng/pipelines/templates/scripts/mashup-doc-index.yml (hunk 2 above)
Comment (chidozieononiwu) [reply eng/pipelines/templates/scripts/mashup-doc-index.yml]: Ill clean this up

**Pull request merged**